### PR TITLE
drivers: eth: mcux: Fix phy setup

### LINKED ISSUE #23221: status register value always reads 0x0000 in eth_mcux_phy_setup
**Describe the bug**
In `drivers/ethernet/eth_mcux.c:eth_mcux_phy_setup` `ENET_ReadSMIData` is called right after calling `ENET_StartSMIRead` without waiting for the MII interrupt, which leads to the value of the status variable to always be `0x0000`.

The correct sequence is documented in the i.MX RT1060 reference manual in chapter 41.7.17.4.

I've hacked the `eth_mcux_phy_setup` function to poll the EIR register until the MII interrupt bit is set before reading the status register and it works (i.e. the content of the status register is correctly retrieved).

**Expected behavior**
Status register value is correctly retrieved.

**Impact**
Calling `eth_mcux_phy_setup` has no effect.

**Environment (please complete the following information):**
- OS: Linux (Ubuntu 18.04)
- Toolchain: Zephyr SDK 0.10.3
 - Commit: 9f7cccc90f648953e0d198fffda22928edb277cf

**Additional context**
This problem occurs on a custom board equipped with the IMXRT1062 MCU but should occur as well with the MIMXRT1060-EVK.

### PR DESCRIPTION
**Pull request by armandciejak**
Fixes #23221.

### CODE REVIEW

**Comment by pfalcon** [comment]
@armandciejak: Thanks for the patch. Previously, we had cases when updates to eth_mcux broke particular boards using it and/or particular usecases. So, can you please describe how you tested these changes?

**Comment by armandciejak** [comment]
@pfalcon I tested it with the MIMXRT1060-EVK board. I logged the value read from the register and checked it against the PHY doc.

I verified that the Ethernet communication is working with the ping command and the DHCP client sample app. I tested with a 100Mbit switch and a 10Mbit hub and checked that link speed is correctly detected.

**Comment by pfalcon** [comment]
> I verified that the Ethernet communication is working with the ping command and the DHCP client sample app. I tested with a 100Mbit switch and a 10Mbit hub and checked that link speed is correctly detected.

Thanks. Can you also please test higher-level protocols, e.g. samples samples/net/sockets/{dumb_http_server, dumb_http_server_mt, big_http_download} ? Would you have a chance to test it on frdm_k64f board too? (I'm going to test it on that board myself too.)

**Comment by armandciejak** [comment]
> Thanks. Can you also please test higher-level protocols, e.g. samples samples/net/sockets/{dumb_http_server, dumb_http_server_mt, big_http_download} ? Would you have a chance to test it on frdm_k64f board too? (I'm going to test it on that board myself too.)

I do not have a frdm_k64f board.

File: drivers/ethernet/eth_mcux.c [hunk 1]
```diff
@@ -412,9 +412,11 @@ static void eth_mcux_phy_setup(void)
 	} else {
 		if (oms_override & PHY_OMS_NANDTREE_MASK) {
 			oms_override &= ~PHY_OMS_NANDTREE_MASK;
-			res = PHY_Write(ENET, phy_addr, PHY_OMS_OVERRIDE_REG, oms_override);
+			res = PHY_Write(ENET, phy_addr, PHY_OMS_OVERRIDE_REG,
+					oms_override);
 			if (res != kStatus_Success) {
-				LOG_WRN("Writing PHY register failed with status 0x%x", res);
```
Comment (tbursztyka) [inline drivers/ethernet/eth_mcux.c]: merge with previous patch and indent properly the lines.

**Comment by armandciejak** [comment]
I tested it with the `dumb_http_server` and the `big_http_download` sample app and it worked fine.

**Review by agansari — CHANGES_REQUESTED** [review]
I would merge all in a single commit.
Please fix checkpatch messages.

**Comment by agansari** [comment]
frdm_k64f (Kinetis SoC) should not be affected, patch only applies to i.MX SoCs.

File: drivers/ethernet/eth_mcux.c (hunk 1 above)
Comment (armandciejak) [reply drivers/ethernet/eth_mcux.c]: Indentation looks fine to me, can you explain me where it needs to be changed.

There are now 3 commits, do you want me to squash all of them into a single one or only squash the 2 last ones?

Comment (tbursztyka) [reply drivers/ethernet/eth_mcux.c]: The last 2.

And indentation at line 419 below is definitely wrong

**Comment by armandciejak** [comment]
> would merge all in a single commit.

I would keep 2 separate commits because each one fix a different problem.

**Review by agansari — APPROVED** [review]

File: drivers/ethernet/eth_mcux.c (hunk 1 above)
Comment (armandciejak) [reply drivers/ethernet/eth_mcux.c]: Can I put everything on a single line? That would means having that line longer that 80 characters.

Comment (jukkar) [reply drivers/ethernet/eth_mcux.c]: > Can I put everything on a single line? That would means having that line longer than 80 characters.

No, try to keep lines shorter than 80. If needed, just split them in suitable places.

Comment (armandciejak) [reply drivers/ethernet/eth_mcux.c]: I shortened the message, it should now be fine.

**Comment by armandciejak** [comment]
Anyone have an idea why *test_k_sleep* fails?

**Comment by jhedberg** [comment]
> Does anyone have an idea why test_k_sleep fails?

Nothing to do with this PR. It's one of the sporadically failing cortex_r5 tests. I've triggered a rerun of the test - hopefully it passes now.

**Review by jukkar — APPROVED** [review]
LGTM

**Pull request merged**